From b1f1009a2f8111ad73dd1970ba07f8b2fa971114 Mon Sep 17 00:00:00 2001
From: hjc <hjc@rock-chips.com>
Date: Tue, 20 Aug 2013 15:47:19 +0800
Subject: [PATCH] rk3026_86v:     1.config lcd for new hardware;     2.fix
 camera lead to reboot;     3.modify arm/logic freq and voltage;

---
 arch/arm/configs/rk3026_86v_defconfig         |  2 +-
 .../arm/mach-rk3026/board-rk3026-86v-camera.c |  4 ++--
 arch/arm/mach-rk3026/board-rk3026-86v.c       |  8 +++----
 arch/arm/mach-rk3026/board-rk3028a-86v.c      | 22 +++++++++----------
 4 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/arch/arm/configs/rk3026_86v_defconfig b/arch/arm/configs/rk3026_86v_defconfig
index 6060b5193783..a87dd60c6e68 100644
--- a/arch/arm/configs/rk3026_86v_defconfig
+++ b/arch/arm/configs/rk3026_86v_defconfig
@@ -307,7 +307,7 @@ CONFIG_FB_ROCKCHIP=y
 CONFIG_DUAL_LCDC_DUAL_DISP_IN_KERNEL=y
 CONFIG_LCDC_RK3188=y
 CONFIG_LCDC0_RK3188=y
-CONFIG_LCD_RK3028_86V=y
+CONFIG_LCD_RK2926_V86=y
 CONFIG_RK_TRSM=y
 CONFIG_RGA_RK30=y
 CONFIG_LOGO=y
diff --git a/arch/arm/mach-rk3026/board-rk3026-86v-camera.c b/arch/arm/mach-rk3026/board-rk3026-86v-camera.c
index 113b08ac5d70..27c7cd2465a2 100755
--- a/arch/arm/mach-rk3026/board-rk3026-86v-camera.c
+++ b/arch/arm/mach-rk3026/board-rk3026-86v-camera.c
@@ -49,7 +49,7 @@ static struct rkcamera_platform_data new_camera[] = {
                         0,
                         1,
                         0),
-                        
+    /*                    
     new_camera_device(RK29_CAM_SENSOR_GC0308,
                         back,
                         RK30_PIN3_PD7,
@@ -57,7 +57,7 @@ static struct rkcamera_platform_data new_camera[] = {
                         0,
                         1,
                         0), 
-                        
+        */                
     new_camera_device_end
 };
 /*---------------- Camera Sensor Macro Define Begin  ------------------------*/
diff --git a/arch/arm/mach-rk3026/board-rk3026-86v.c b/arch/arm/mach-rk3026/board-rk3026-86v.c
index fe058638bc6c..391babd16bd0 100755
--- a/arch/arm/mach-rk3026/board-rk3026-86v.c
+++ b/arch/arm/mach-rk3026/board-rk3026-86v.c
@@ -1629,10 +1629,10 @@ static void __init rk30_reserve(void)
 *	clock
 ************************************************************/
 static struct cpufreq_frequency_table dvfs_arm_table[] = {
-	//{.frequency = 312 * 1000,       .index = 1200 * 1000},
-	//{.frequency = 504 * 1000,       .index = 1200 * 1000},
-	{.frequency = 816 * 1000,       .index = 1250 * 1000},
-	//{.frequency = 1008 * 1000,      .index = 1200 * 1000},
+	{.frequency = 312 * 1000,       .index = 1150 * 1000},
+	{.frequency = 504 * 1000,       .index = 1150 * 1000},
+	{.frequency = 816 * 1000,       .index = 1200 * 1000},
+	{.frequency = 1008 * 1000,      .index = 1250 * 1000},
 	//{.frequency = 1200 * 1000,      .index = 1200 * 1000},
 	//{.frequency = 1416 * 1000,      .index = 1200 * 1000},
 	//{.frequency = 1608 * 1000,      .index = 1200 * 1000},
diff --git a/arch/arm/mach-rk3026/board-rk3028a-86v.c b/arch/arm/mach-rk3026/board-rk3028a-86v.c
index 2c473a421057..6a15c15ac38f 100755
--- a/arch/arm/mach-rk3026/board-rk3028a-86v.c
+++ b/arch/arm/mach-rk3026/board-rk3028a-86v.c
@@ -1584,10 +1584,10 @@ static void __init rk30_reserve(void)
 *	clock
 ************************************************************/
 static struct cpufreq_frequency_table dvfs_arm_table[] = {
-	{.frequency = 312 * 1000,       .index = 1200 * 1000},
-	{.frequency = 504 * 1000,       .index = 1200 * 1000},
-	{.frequency = 816 * 1000,       .index = 1200 * 1000},
-	//{.frequency = 1008 * 1000,      .index = 1200 * 1000},
+	{.frequency = 312 * 1000,       .index = 1100 * 1000},
+	{.frequency = 504 * 1000,       .index = 1100 * 1000},
+	{.frequency = 816 * 1000,       .index = 1150 * 1000},
+	{.frequency = 1008 * 1000,      .index = 1200 * 1000},
 	//{.frequency = 1200 * 1000,      .index = 1300 * 1000},
 	//{.frequency = 1416 * 1000,      .index = 1200 * 1000},
 	//{.frequency = 1608 * 1000,      .index = 1200 * 1000},
@@ -1595,18 +1595,18 @@ static struct cpufreq_frequency_table dvfs_arm_table[] = {
 };
 
 static struct cpufreq_frequency_table dvfs_gpu_table[] = {
-	{.frequency = 100 * 1000,       .index = 1200 * 1000},
+	//{.frequency = 100 * 1000,       .index = 1200 * 1000},
 	{.frequency = 200 * 1000,       .index = 1200 * 1000},
-	{.frequency = 266 * 1000,       .index = 1200 * 1000},
-	{.frequency = 300 * 1000,       .index = 1200 * 1000},
-	{.frequency = 400 * 1000,       .index = 1200 * 1000},
+	//{.frequency = 266 * 1000,       .index = 1200 * 1000},
+	//{.frequency = 300 * 1000,       .index = 1200 * 1000},
+	//{.frequency = 400 * 1000,       .index = 1200 * 1000},
 	{.frequency = CPUFREQ_TABLE_END},
 };
 
 static struct cpufreq_frequency_table dvfs_ddr_table[] = {
-	{.frequency = 200 * 1000 + DDR_FREQ_SUSPEND,    .index = 1200 * 1000},
-	{.frequency = 300 * 1000 + DDR_FREQ_VIDEO,      .index = 1200 * 1000},
-	{.frequency = 400 * 1000 + DDR_FREQ_NORMAL,     .index = 1200 * 1000},
+	//{.frequency = 200 * 1000 + DDR_FREQ_SUSPEND,    .index = 1200 * 1000},
+	//{.frequency = 300 * 1000 + DDR_FREQ_VIDEO,      .index = 1200 * 1000},
+	{.frequency = 360 * 1000 + DDR_FREQ_NORMAL,     .index = 1200 * 1000},
 	{.frequency = CPUFREQ_TABLE_END},
 };
 
-- 
2.35.3

